src/logging.c

```diff
@@ -749,6 +792,10 @@ rcutils_ret_t rcutils_logging_shutdown(void)
   }
   g_num_log_msg_handlers = 0;
   g_rcutils_logging_initialized = false;
+
+  if (g_consol_mode_modified){
+    SetConsoleMode(GetStdHandle(STD_ERROR_HANDLE), g_original_console_mode);
+  }
```

Comment:
these seem the Windows specific code? probably we do not need to have this `if` statement for Non-Windows platform at all, so are the static global variables?

this actually failed to build with Linux,

```bash
Finished <<< rosidl_generator_dds_idl [0.36s]
--- stderr: rcutils                                  
/root/ros2_ws/colcon_ws/src/ros2/rcutils/src/logging.c:90:8: error: unknown type name ‘DWORD’
   90 | static DWORD g_original_console_mode = 0;
      |        ^~~~~
/root/ros2_ws/colcon_ws/src/ros2/rcutils/src/logging.c: In function ‘rcutils_logging_shutdown’:
/root/ros2_ws/colcon_ws/src/ros2/rcutils/src/logging.c:797:5: warning: implicit declaration of function ‘SetConsoleMode’ [-Wimplicit-function-declaration]
  797 |     SetConsoleMode(GetStdHandle(STD_ERROR_HANDLE), g_original_console_mode);
      |     ^~~~~~~~~~~~~~
/root/ros2_ws/colcon_ws/src/ros2/rcutils/src/logging.c:797:20: warning: implicit declaration of function ‘GetStdHandle’ [-Wimplicit-function-declaration]
  797 |     SetConsoleMode(GetStdHandle(STD_ERROR_HANDLE), g_original_console_mode);
      |                    ^~~~~~~~~~~~
/root/ros2_ws/colcon_ws/src/ros2/rcutils/src/logging.c:797:33: error: ‘STD_ERROR_HANDLE’ undeclared (first use in this function)
  797 |     SetConsoleMode(GetStdHandle(STD_ERROR_HANDLE), g_original_console_mode);
      |                                 ^~~~~~~~~~~~~~~~
/root/ros2_ws/colcon_ws/src/ros2/rcutils/src/logging.c:797:33: note: each undeclared identifier is reported only once for each function it appears in
gmake[2]: *** [CMakeFiles/rcutils.dir/build.make:222: CMakeFiles/rcutils.dir/src/logging.c.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:319: CMakeFiles/rcutils.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake: *** [Makefile:146: all] Error 2
---
Failed   <<< rcutils [2.08s, exited with code 2]
```

Reply:
Ah sorry, I forgot to check again if it still builds on Linux after fixing the Windows version. I fixed it in the last commit.